cuda_next/include/detail/dimensions.hpp

```diff
@@ -0,0 +1,109 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Next in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDA_NEXT_DETAIL_DIMENSIONS
+#define _CUDA_NEXT_DETAIL_DIMENSIONS
+
+#include <cuda/std/mdspan>
+
+namespace cuda_next
+{
+
+template <typename T, size_t... Extents>
+using dimensions = ::cuda::std::extents<T, Extents...>;
+
+// not unsigned because of a bug in ::cuda::std::extents
+using index_type = int;
```

Comment:
I am a bit scared by that alias in the `cuda_next` namespace

Reply:
Renamed to dimensions_index_type